### PR TITLE
game over check on block spawn

### PR DESCRIPTION
**Pull request by Daniel6**
If the game tries to spawn a block but there is already a block in the spawn location, the game ends.

There are still some bugs in the shutdown procedure where you lose the ability to see input in that terminal, see Trello.

### CODE REVIEW

File: src/block_factory.c [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
+        s->running = 0;
+    }
+}
+
+/*
+ * Make sure that all cells occupied by the block are empty.
+ * Perform this check immediately after spawning a block so that blocks do not
+ * spawn in eachother.
+ */
+int spawnSpaceAvailable(State* s) {
```
Comment (thecardkid) [inline src/block_factory.c:28]: oops i kinda screwed up by using camel case in this file. can you change everything to underscore notation?

File: src/utils.h [hunk 2]
```diff
@@ -65,6 +65,7 @@ typedef struct {
    int grid[GRID_W][GRID_H];
    int score;
    int level;
+   int running; // 1: Game in progress 0: Game over
```
Comment (thecardkid) [inline src/utils.h:68]: Can you make this an enum so we can add `pause` and `bossmode` states later

Comment (thecardkid) [reply src/utils.h:68]: and maybe rename it?

**Review by thecardkid — CHANGES_REQUESTED** [review]

File: src/utils.h (hunk 2 above)
Comment (Daniel6) [reply src/utils.h:68]: im going to roll this change into an upcoming PR

**Comment by Daniel6** [comment]
rolled into PR#19

**Pull request closed without merging**